scripts/register_member.sh

```diff
@@ -0,0 +1,156 @@
+    esac
+done
+
+# Validate parameters
+if [[ -z $network_url ]]; then
+    echo "Missing parameter --network_url"
+    exit 1
+elif [[ -z $certificate_dir ]]; then
+    echo "Missing parameter --certificate_dir"
+    exit 1
+elif [[ -z $akv_kid ]]; then
+    echo "Missing parameter --akv_kid"
+    exit 1
+elif [[ -z $akv_authorization ]]; then
+    echo "Missing parameter --akv_authorization"
+    exit 1
+fi
+
+# Check if 'certificates' is in the URL
+if [[ $akv_kid != *"certificates"* ]]; then
+    echo "Error: The URL does not contain 'certificates'."
+    exit 1
+fi
+
+# Variables
+akv_keys_url="${akv_kid/certificates/keys}"
+api_version="7.1"
+signature_file="vol/signature"
+created_at=`date -uIs`
+member_name=$(echo "$AKV_KID" | awk -F'/' '{print $(NF-1)}')
+signing_cert="$certificate_dir/${member_name}_cert.pem"
+cose_sign1_file="vol/cose_sign1"
+ccf_services_cert="$certificate_dir/service_cert.pem"
+fingerprint=$(openssl x509 -noout -fingerprint -sha256 -inform pem -in "$signing_cert" | sed 's/://g' | awk -F= '{print $2}' | tr '[:upper:]' '[:lower:]')
+state_digest_url="$network_url/gov/members/state-digests/${fingerprint}"
+state_digest_update_url="$state_digest_url:update?api-version=2024-07-01"
+state_digest_ack_url="$state_digest_url:ack?api-version=2024-07-01"
+tbs="/tmp/tbs"
+
+echo "Update ledget state digest"
+echo "create at: " $created_at
+echo "State digest ack url: " $state_digest_ack_url
+echo "AKV Signing Url: " $akv_keys_url
+echo "Signing Cert: " $signing_cert
+echo "Fingerprint: " $fingerprint
+
+# Prepare signature
+rm -f $tbs
+ccf_cose_sign1_prepare --ccf-gov-msg-type proposal --ccf-gov-msg-created_at $created_at --content vol/empty_file --signing-cert $signing_cert >$tbs
+echo "AKV update state digest signature prepared"
+cat $tbs | jq
+
+# Perform the curl request to sign the data
+
+rm -f $signature_file
+curl -s -X POST "$akv_keys_url/sign?api-version=$api_version" \
+  --data @$tbs \
+  -w "\n" \
+  -H "Authorization: $akv_authorization" \
+  -H "Content-Type: application/json" > $signature_file
+echo "AKV update state digest signature retrieved"
+cat $signature_file | jq
+
+# Check if the signature was retrieved successfully
+if [ ! -s $signature_file ]; then
+    echo "Failed to retrieve signature"
+    exit 1
+fi
+echo "Start finishing the update state digest signature"
+
+#  update and retrieve the latest state digest
+ccf_cose_sign1_finish \
+  --ccf-gov-msg-type state_digest  \
+  --ccf-gov-msg-created_at $created_at \
+  --content vol/empty_file \
+  --signing-cert $signing_cert \
+  --signature $signature_file \
+  | curl $state_digest_update_url \
+  -X POST \
+  --cacert $ccf_services_cert  \
+  --data-binary @- \
+   -H "content-type: application/cose" \
+   --cert $signing_cert \
+   --silent \
+  | jq
+
+state_digest=$(curl $KMS_URL/gov/members/state-digests/385be3de9ae7d8874636f73f2d273ee44f83847f0e15a238c02345cf94b230a7?api-version=2024-07-01 --cacert ${KEYS_DIR}/service_cert.pem | jq)
```

Comment:
Is ok to have this value 385be3de9ae7d8874636f73f2d273ee44f83847f0e15a238c02345cf94b230a7 here?

Reply:
done